src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/NetCorePbkdf2Provider.cs

```diff
@@ -54,11 +55,11 @@ private static byte[] DeriveKeyImpl(string password, byte[] salt, KeyDerivationP
                     throw new ArgumentOutOfRangeException();
             }
 
-            var passwordBytes = Encoding.UTF8.GetBytes(password);
-            using (var rfc = new Rfc2898DeriveBytes(passwordBytes, salt, iterationCount, algorithmName))
-            {
-                return rfc.GetBytes(numBytesRequested);
-            }
+            var maxBytes = Encoding.UTF8.GetMaxByteCount(password.Length);
+            Span<byte> bytes = (maxBytes > 256) ? ArrayPool<byte>.Shared.Rent(maxBytes) : stackalloc byte[256];
```

Comment:
I'm not sure if this turns it in to an Oracle since stackalloc is faster than the ArrayPool. Also, I just recalled @GrabYourPitchforks recommending against using the shared array pool for crypto things.

Reply:
@pranavkm how do you feel about if we just do stackalloc for up to 256 and use the old alloc from GetBytes for crazy long passwords (which has to be extremely uncommon)

Reply:
Chatted with @pranavkm offline, we will put this into the bucket of spanification things to revisit in a separate PR,